source/common/regular-expressions.js

```diff
@@ -14,6 +14,190 @@
+      'g' + flag)
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFnRE': function () {
+    return RegExp(
+      /(\[\^([\da-zA-Z_-]+)\][^:]|\[\^([\da-zA-Z_-]+)\]$)/.source,
+      'g')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFnReferenceRE': function () {
+    return RegExp(
+      /^\[\^.+\]:\s/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @param   {boolean}  multiline  Whether to match multiline
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFnRefRE': function (multiline = false) {
+    let flag = (multiline) ? 'm' : ''
+    return RegExp(
+      /^\[\^([\da-zA-Z_-]+)\]: (.+)/.source,
+      'g' + flag)
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFootnoteRefRE': function () {
+    return RegExp(
+      /\[\^[^\]]+\]/.source,
+      'i')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown Headings.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getHeadRE': function () {
+    return RegExp(
+      /^(#{1,6}) (.*)/.source,
+      'g')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown Headings.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getHeadingRE': function () {
+    return RegExp(
+      /(#+)\s+/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown highlighting (???).
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getHighlightRE': function () {
+    return RegExp(
+      /::.+?::|==.+?==/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches file IDs as in the settings
+   *
+   * @param   {boolean}  multiline  Whether to match multiline
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getIDRE': function (multiline = false) {
+    let flag = (multiline) ? 'm' : ''
+    let idRegExpString = global.config.get('zkn.idRE')
+    // Make sure the ID definitely has at least one
+    // capturing group to not produce errors.
+    if (!(/\(.+?\)/.test(idRegExpString))) {
+      idRegExpString = `(${idRegExpString})`
+    }
+
+    return RegExp(
+      idRegExpString,
+      'g' + flag
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches inline frames.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getIframeRE': function () {
+    return RegExp(
+      /^<iframe.*?>.*?<\/iframe>$/.source,
```

Comment:
Note to myself: We definitely need a better way to get iFrames, because right now it's still that users can actually open a door to malware, because this will just load *any* iFrame, not just "good" ones (e.g. YouTube, Vimeo, etc.)